yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryConfig.java

```diff
@@ -225,6 +224,9 @@ void checkResultInternal(@Nonnull Object actual, @Nonnull String queryDescriptio
                 var success = isExact ? getVal().equals(actualPlan) : actualPlan.contains((String) getVal());
                 if (success) {
                     logger.debug("✅️ plan match!");
+                    while (resultSet.next()) {
```

Comment:
Why do you need to exhaust the resultSet here?

Reply:
`resultSet` is actually a reference to `actual` parameter, and by the call stack, (see `QueryExecutor:98`):
```
                    config.checkResult(queryResult, this.toString());
                    if (queryResult instanceof RelationalResultSet) {
                        continuationAfter = ((RelationalResultSet) queryResult).getContinuation();
                    }
```

will fail otherwise.

Reply:
Yes, but why would there be more results? Shouldn't an explain always return one?

Reply:
Good catch. There was a bug in the resultSet's `hasNext()` that prevented from returning the continuation when we were on the last row. This is fixed and now covered by the improved `testMidContinuationFails` test.